mailpoet/lib/Automation/Engine/Migrations/Migrator.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Migrations;
+
+use MailPoet\Automation\Engine\Exceptions;
+use wpdb;
+
+class Migrator {
+  /** @var string */
+  private $prefix;
+
+  /** @var wpdb */
+  private $wpdb;
+
+  public function __construct() {
+    global $wpdb;
+    $this->prefix = $wpdb->prefix . 'mailpoet_automation_';
+    $this->wpdb = $wpdb;
```

Comment:
This will most likely break on some hosts with advanced object caches (like WPE) and I think also in conjunction with plugins like W3 Total cache. I never really digged deep enough into why, but it seems, that sometimes $wpdb gets replaced by a new $wpdb later in the process and then we can't connect to the database anymore. We had the problem in the [CentralStock plugin](https://woocommerce.com/products/centralstock-for-woocommerce/) and I had it also with a client plugin. I injected a WpdbFactory instead, which always fetched the global `$wpdb` when it was needed.

Reply:
@websupporter Oh, well... thanks, that's a good point. I have a note that we should write a wrapper class for WPDB anyway so we can handle query errors (and convert them to exceptions) so this is something we should handle in there as well.

Reply:
Added this to the checklist.